### PR TITLE
Update notebook converter

### PR DESCRIPTION
**Pull request by ndmlny-qs**
### Motivation

Updating the notebook converter for easier future refactors, and to accommodate the diagnostics tools.

### Changes proposed

- Coin flipping tutorial was reverted back to the main branch.
- A new CSS file for Bokeh Application (diagnostic tools) for
  rendering in Docusaurus.
- Updated Plotting.jsx to include the new Bokeh CSS.
- Heavily modified the notebook converter for easier future refactoring.
  The required number of changes to handle the diagnostics tools
  necessitated a refactor, not just for ease of reading, but also for
  easier additions to how we handle different cell outputs.
- The modified converter removes HTML comments, fixes a few missing
  outputs from tutorials, and updates docstrings to conform to what is
  used in the code base.

### Test Plan

The website was built locally and visually inspected against beanmachine.org for completeness.

### Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **[CONTRIBUTING](https://github.com/facebookresearch/beanmachine/blob/main/CONTRIBUTING.md)** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by feynmanliang** [comment]
```
  File "/data/sandcastle/boxes/eden-trunk-hg-fbcode-fbsource/fbcode/buck-out/dev/gen/aab7ed39/beanmachine/website/scripts/ipynb-to-mdx#link-tree/beanmachine/website/scripts/convert_ipynb_to_mdx.py", line 501, in handle_pandas
    mdx = md_df.to_markdown(index=False)
  File "/data/sandcastle/boxes/eden-trunk-hg-fbcode-fbsource/fbcode/buck-out/dev/gen/aab7ed39/beanmachine/website/scripts/ipynb-to-mdx#link-tree/pandas/core/frame.py", line 2018, in to_markdown
    result = tabulate.tabulate(self, **kwargs)
```

File: website/scripts/convert_ipynb_to_mdx.py [hunk 1]
```diff
@@ -30,526 +31,779 @@
...
+            the tuple is the MDX formatted string.
+    """
+    output = sorted(values, key=lambda item: item["index"])
+    index = output[0]["index"]
+    md = "\n".join([item["data"] for item in output if item["data"]])
+    return [(index, f"<CellOutput>\n{{\n  `{md}`\n}}\n</CellOutput>\n\n")]
 
-    # Begin to build the mdx string.
+
+def handle_cell_outputs(  # noqa: C901 (flake8 too complex)
```
Comment (horizon-blue) [inline website/scripts/convert_ipynb_to_mdx.py:587]: Our internal linter still insist that this function is too complex. Can we break it down into a few smaller ones?

File: website/scripts/convert_ipynb_to_mdx.py [hunk 2]
```diff
@@ -30,526 +31,779 @@
...
+    ]
+    prioritized_cell_output_dtypes = [
+        sorted(
+            set(dtypes).intersection(set(priorities)),
+            key=lambda dtype: priorities.index(dtype),
+        )
+        for dtypes in cell_output_dtypes
+    ]
+    plotly_flags = [
+        any(map(lambda output: "plotly" in output, outputs))
```
Comment (horizon-blue) [inline website/scripts/convert_ipynb_to_mdx.py:642]: Similar to the other place, can we rewrite this with list comprehension as well?

File: website/src/css/bokeh.css [hunk 3]
```diff
@@ -0,0 +1,26 @@
+.bk.bm-tool-loading {
```
Comment (horizon-blue) [inline website/src/css/bokeh.css:1]: Can we add the license to the top of this file?

**Comment by horizon-blue** [comment]
Thanks for the update! Since the marginal 1D widget is already landed, can we update any one of the tutorial to use the marginal widget in this PR, so we can see how it's going to look like on docusaurus?

**Comment by ndmlny-qs** [comment]
I've updated the coin flipping tutorial to include the 1D marginal tool.

**Pull request closed without merging**